staging/src/k8s.io/apiserver/pkg/storage/cacher/lister_watcher.go

```diff
@@ -59,7 +63,11 @@ func (lw *listerWatcher) List(options metav1.ListOptions) (runtime.Object, error
 		Predicate:            pred,
 		Recursive:            true,
 	}
-	if err := lw.storage.GetList(context.TODO(), lw.resourcePrefix, storageOpts, list); err != nil {
+	ctx := context.Background()
+	if lw.contextMetadata != nil {
```

Comment:
Hmm, the if condition for the feature above passes in an empty map. So, can we check the number of items in the map as well here to check before creating `metadata.NewOutgoingContext`?